torch/quantization/adaround_fake_quantize.py

```diff
@@ -0,0 +1,102 @@
+import torch
+import torch.nn.init as init
+import math
+from torch.quantization.fake_quantize import FakeQuantize
+from torch.quantization.observer import MovingAverageMinMaxObserver
+from torch.quantization.qconfig import *
+from torch.quantization.fake_quantize import *
+
+
+class AdaRoundFakeQuantize(FakeQuantize):
+    ''' This class is an extension of the FakeQuantize module and is used similarly.
+    The difference between the two is that you can use the continous_V attribute to alter the
+    default rounding scheme during quantization. The defualt is to round to the nearest number,
+    but applying an optimizer on this attribute can yield a better rounding scheme and improve
+    quantization results.
+
+    '''
+    adaround_default_attribute_settings = {'continous_V': None,
+                                           'beta_high': 8,
+                                           'beta_low': 2,
+                                           'norm_scaling': 10,
+                                           'regularization_scaling': .1}
+
+    def __init__(self, *args, **keywords):
+        settings = AdaRoundFakeQuantize.adaround_default_attribute_settings
+
+        for attribute in settings:
+            if attribute in settings:
+                setattr(self, attribute, keywords[attribute])
+                del keywords[attribute]
+            else:
+                setattr(self, attribute, settings[attribute])
+
+        self.tuning = False
+        super(AdaRoundFakeQuantize, self).__init__(*args, **keywords)
+
+    def forward(self, X):
+        if self.continous_V is None:
+            # need some value already assigned to continuous_V for kaiming to work
+            self.continous_V = torch.nn.Parameter(torch.zeros(X.size()))
+            init.kaiming_uniform_(self.continous_V, a=math.sqrt(5))
```

Comment:
this should be done in __init__?

Reply:
can't be done in init because you need to know X.size() (which is unknown when init is called), to get around that i set continous_V to None and the first time the forward is run we get X.size() and its updated accordingly

Reply:
oh, I see, please add a comment